### PR TITLE
Added sst-core and variant for core to sst-macro.

### PR DESCRIPTION
**Pull request by sknigh**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    depends_on('python@:2')
+    depends_on('zlib', type='build')
+    depends_on('mpi', when='+mpi')
+    depends_on('boost@1.56.0:', type='build', when='+boost')
+
+    def configure_args(self):
+        args = []
+        spec = self.spec
+
+        if '~mpi' in spec:
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/sst-core/package.py]: can this also be explicit and `--enable-mpi` when `+mpi` is provided?

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            args.append('--disable-mpi')
+
+        if '+boost' in spec:
+            args.append('--with-boost=%s' % spec['boost'].prefix)
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/sst-core/package.py]: likewise this should `--disable-boost` when boost is off

File: var/spack/repos/builtin/packages/sst-macro/package.py [hunk 3]
```diff
@@ -90,6 +93,9 @@ def configure_args(self):
             if '+skeletonizer' in spec:
                 args.append('--with-clang=' + spec['llvm'].prefix)
 
+        if '+core' in spec:
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/sst-macro/package.py]: what happens with `~core`?  I'm not familiar with how these packages are related so I can't say what to do with this one.

Comment (sknigh) [reply var/spack/repos/builtin/packages/sst-macro/package.py]: Core is optional substitute for the PDES (Parallel discreet event simulation) engine, I agree that the naming is confusing.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 2 above)
Comment (sknigh) [reply var/spack/repos/builtin/packages/sst-core/package.py]: ```--with-boost[=DIR]``` is the only documented boost flag. Boost is an optional dependency.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 1 above)
Comment (sknigh) [reply var/spack/repos/builtin/packages/sst-core/package.py]: The build will fail unless an MPI compiler wrapper is found, or ```--disable-mpi``` flag is used. ```--enable-mpi``` is not documented in the configure script and gets ignored in the configure step.

Comment (tgamblin) [reply var/spack/repos/builtin/packages/sst-core/package.py]: ok I guess omit that, then.  The worry is that one day `--disable-mpi` will become the default and `+mpi` won't be `+mpi`.  Seems unlikely here.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 2 above)
Comment (tgamblin) [reply var/spack/repos/builtin/packages/sst-core/package.py]: If it's standard `autoconf`, `--with` flags always have `--without` and `--disable` ones should have a corresponding `--enable` (even if it is not documented).  I think that's just how `AC_ARG_WITH` and `AC_ARG_ENABLE` work.  So I guess I don't see the harm in being explicit?  If it's not standard autotools I guess we don't have many options.

File: var/spack/repos/builtin/packages/sst-macro/package.py (hunk 3 above)
Comment (tgamblin) [reply var/spack/repos/builtin/packages/sst-macro/package.py]: oh.

**Comment by sknigh** [comment]
@tgamblin Thanks. v8.1.0 will be released in a few days, I will take a closer look at inverse flags and add them to the package if they work.

**Pull request merged**